File: Sistemi operativi/ContoBancario.py
class contoBancario:
    def __init__(self, nA=0, nomeProp="", saldo=0, saldoMinimo=100):
        self.nA = nA
        self.nomeProp = nomeProp
        self.saldo = saldo
        self.saldoMinimo = saldoMinimo
    
    def copy(self):
        return contoBancario(self.nA, self.nomeProp, self.saldo, self.saldoMinimo)

    def getNA(self):
        return self.nA
    
    def getNomeProp(self):
        return self.nomeProp
    
    def getSaldo(self):
        return self.saldo
    
    def getSaldoMinimo(self):
        return self.saldoMinimo
    
    def setNA(self, nA):
        self.nA = nA
    
    def setNomeProp(self, nomeProp):
        self.nomeProp = nomeProp
    
    def setSaldo(self, saldo):
        self.saldo = saldo
    
    def deposita(self, importo):
        if importo > 0:
            self.saldo += importo
            print("Deposito di: " + str(importo) + " effettuato. Saldo attuale: " + str(self.saldo))
        else:
            print("Importo non valido")
    
    def preleva(self, importo):
        if importo <= self.saldo - self.saldoMinimo:
            self.saldo -= importo
            print("Prelievo di: " + str(importo) + " effettuato con successo. Saldo attuale: " + str(self.saldo))
        else:
            print("Fondi insufficienti")


nA = int(input("Inserisci numero account: "))
nomeProp = input("Inserisci nome account: ")

conto = contoBancario(nA, nomeProp)

n = int(input("Inserisci 1 per depositare e 2 per prelevare: "))
if n == 1:
    importo = int(input("Inserisci importo da depositare: "))
    conto.deposita(importo)
else:
    importo = int(input("Inserisci importo da prelevare: "))
    conto.preleva(importo)

File: C++/ContoBancario.cpp
#include <iostream>
using namespace std;

class ContoBancario{
    private:
        int numeroConto;
        double saldo;
    public:
        ContoBancario(){
            numeroConto = 0;
            saldo = 0.0;
        }
        ContoBancario(ContoBancario& cb){
            numeroConto = cb.numeroConto;
            saldo = cb.saldo;
        }
        ContoBancario(int nc, double s){
            numeroConto = nc;
            saldo = s;
        }

        void setNumeroConto(int nc){
            numeroConto = nc;
        }
        void setSaldo(int s){
            saldo = s;
        }
        
        int getNumeroConto(){
            return numeroConto;
        }
        int getSaldo(){
            return saldo;
        }
        
        void deposita(int denaro){
            saldo += denaro;
            cout<<"Hai depositato: "<< denaro;
        }
        void preleva(int prelievo){
            if (saldo <= prelievo){
                cout<<"Denaro insufficente";
            }
            else{
                saldo -= prelievo;
            }
        }
        void stampaSaldo(){
            cout<<", "<<"Saldo attuale: " << saldo;
        }
};

int main(){
    ContoBancario* conto = new ContoBancario[1];
    int nc;
    double s;
    cout<<"Inserisci numero conto e saldo per entrare: "<<endl;
    cin>> nc >> s;
    conto[0] = ContoBancario(nc, s);
    string op;
    cout<<"Inserisci -deposito se vuoi depositare, altrimenti -prelievo per prelevare: ";
    cin>> op;
    if(op == "deposito"){
        cout<<"Inserisci quanto vuoi depositare: ";
        double dep;
        cin>> dep;
        conto[0].deposita(dep);
        conto[0].stampaSaldo();
    }else if(op == "prelievo"){
        cout<<"Inserisci quanto vuoi prelevare: ";
        double prel;
        cin>> prel;
        conto[0].preleva(prel);
        conto[0].stampaSaldo();
    }
    return 0;
}

